java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java

```diff
@@ -114,6 +114,25 @@ private void validateDateVector(ValueVector vector, DateUnit expectedDateUnit) {
         "Expecting date unit %s, actual date unit %s.", expectedDateUnit, dateType.getUnit());
   }
 
+  private void validateDecimalVector(ValueVector vector) {
```

Comment:
is this only testable via the fuzz files, or can a unit test be added?

Reply:
A unit test added. Thanks for the good suggestion.